### PR TITLE
Changed pi_check to handle parameter values where possible

### LINKED ISSUE #3868: pi_check does not seems to be running over the values in bind_parameters
After #2808, the visualizations represent fractions of `pi` very nicely. 
 
```python
my_u2_circuit = QuantumCircuit(1, name='my_u2')
phi = Parameter('phi')
lam = Parameter('lambda')
my_u2_circuit.u3(3.141592653589793, phi, lam, 0)
print(my_u2_circuit)
```
```
        ┌───────────────────┐
q_0: |0>┤ U3(pi,phi,lambda) ├
        └───────────────────┘
```

However, there are some corner cases missing. Like this one:
```python
my_u2 = my_u2_circuit.to_gate()
qr = QuantumRegister(1, name='qr')
circuit = QuantumCircuit(qr, name='circuit')
circuit.append(my_u2, [qr[0]])
circuit = circuit.bind_parameters({phi: 3.141592653589793, lam: 3.141592653589793})
print(circuit)
```
```
         ┌──────────────────────────────────────────┐
qr_0: |0>┤ My_u2(3.14159265358979,3.14159265358979) ├
         └──────────────────────────────────────────┘
```
## What is expected?
The last example should look like this:
```
         ┌──────────────┐
qr_0: |0>┤ My_u2(pi,pi) ├
         └──────────────┘
```

### PR DESCRIPTION
**Pull request by JamesSeaward**
Fixes #3868 

Changed pi_check to handle parameter values where possible and corrected display of numbers that are just less than pi

### Summary
Fix for [https://github.com/Qiskit/qiskit-terra/issues/3868](https://github.com/Qiskit/qiskit-terra/issues/3868) where bound parameters were not displaying correctly as a ratio of `pi`.
Also fixed a problem I noticed in testing where values just less than an integer ratio of `pi` were not displayed correctly.

Test code:
```
from qiskit import *
from qiskit.circuit import *
my_u2_circuit = QuantumCircuit(1, name='my_u2')
phi = Parameter('phi')
lam = Parameter('lambda')
my_u2_circuit.u3(3.141592653589793, phi, lam, 0)
print(my_u2_circuit)
my_u2 = my_u2_circuit.to_gate()
qr = QuantumRegister(1, name='qr')
circuit = QuantumCircuit(qr, name='circuit')
circuit.append(my_u2, [qr[0]])
circuit = circuit.bind_parameters({phi: 3.141592653589793, lam: 3.141592653589793})
print(circuit)
```
now correctly prints:
```
        ┌───────────────────┐
q_0: |0>┤ U3(pi,phi,lambda) ├
        └───────────────────┘
         ┌──────────────┐
qr_0: |0>┤ My_u2(pi,pi) ├
         └──────────────┘
```
### Details and comments
The existing `pi_check` immediately returns the `str` representation of the input if it is of type `ParameterExpression`. This works well before `bind_parameters` is called.

The fix is to try and convert this string representation to a `float` and pass this in to `pi_check` for re-processing.

There was also a bug where values just less than an integer ratio were not being found because `%` was being used: `2.9999... > 3 - eps` but `2.9999... % 1 = 0.9999...` which is obviously much larger than `eps`. 

I don't think a change like this would require a documentation update, but maybe an extra test?
I'm happy to add if needed.

### CODE REVIEW

**Comment by 1ucian0** [comment]
Great work @JamesSeaward  ! Please add a check while I review the PR. Thanks!

File: qiskit/visualization/tools/pi_check.py [hunk 1]
```diff
@@ -45,7 +45,10 @@ def pi_check(inpt, eps=1e-6, output='text', ndigits=5):
         QiskitError: if output is not a valid option.
     """
     if isinstance(inpt, ParameterExpression):
-        return str(inpt)
+        try:
+            return pi_check(float(str(inpt)), eps=eps, output=output, ndigits=ndigits)
```
Comment (kdk) [inline qiskit/visualization/tools/pi_check.py]:
```suggestion
            return pi_check(float(inpt), eps=eps, output=output, ndigits=ndigits)
```
If you're looking to check if the Parameter is fully bound, you can call `float()` on it directly (which will raise a `TypeError` if it is not fully bound).

**Comment by 1ucian0** [comment]
I added tests for this function in b28b05c and I could not reproduce the bug you are mentioning here:
> There was also a bug where values just less than an integer ratio were not being found because % was being used: 2.9999... > 3 - eps but 2.9999... % 1 = 0.9999... which is obviously much larger than eps.

Do you mind extending with a case that fails in master and passes in this branch? Thanks!

**Comment by 1ucian0** [comment]
ping @JamesSeaward

**Comment by 1ucian0** [comment]
Test added in fdcc011

**Pull request merged**